### ISSUE TITLE
feat: use API types from Notion library

### ISSUE THREAD

**Issue by EndBug** [instantish/martian#20]
<sub>[Discuss in #**public-issue-20--feat-use-generics-in-block-interface**](https://slack.com/app_redirect?team=TJ5G67VHU&channel=C0381P3HWEQ)</sub>

Using a generic in the definition of the `Blocks` interface would allow us to better enforce supported block types and get more accurate typings

**Comment by EndBug** [comment]
Actually, while looking into the issue I saw that API types are still exported by `@notionhq/client` (latest version), they're just more difficult to access.

For example, the type we're currently calling `Block` can be exposed like this:

```import {AppendBlockChildrenParameters} from '@notionhq/client/build/src/api-endpoints';
export type Block = AppendBlockChildrenParameters['children'][number];```
We could try to use the existing types from the Notion library (which are way more accurate) instead of our generic ones.

After starting to add these changes I saw that some errors were popping up regarding stuff that has been fixed in #19 and #21, so maybe it's better to wait for those two PRs to get merged before messing with the types

**Comment by marissamarym** [comment]
just merged #19 and #21. I think it definitely makes sense to use notion’s types than our own 👍

**Comment by EndBug** [comment]
Awesome, thanks!
